testing.java

```diff
@@ -0,0 +1,42 @@
+import static org.junit.Assert.assertEquals;
+
+import java.awt.List;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params. provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+ 
+public class testing {
+
+	@ParameterizedTest
+	@MethodSource("stringProvider")
+
+	void test(double num , String[] params) {
+		double result = -1 ;
+		try {
+			result = AreaCalculator.calculate(params) ;
+		} catch (Exception e) {
+			e.printStackTrace();
+		}

```

Comment:
And what about the cases where you throw an exception?
Part of testing is not just testing the happy flow; its also testing where it can go wrong, to make sure you covered all those cases. 
The reason: if you don't, then maybe one day you'll make a change which will not ruin any happy-flow, but will ruin a bad-flow, meaning you'll have a bug waiting without a handler in your code for it.

Reply:
half of them throw an exception and printed in the console.